### PR TITLE
Issue 6502 main

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This is a cherry-pick from 4-2-stable with minor necessary changes.

It has passed all CI and unit tests.

### CODE REVIEW

**Comment by korydraughn** [comment]
@JustinKyleJames Please highlight what has changed between this PR and the cherry-pick.

**Comment by alanking** [comment]
Some personal bookkeeping...

Cherry-picked from...
#6899
#6910
#7003

File: plugins/resources/src/compound.cpp [hunk 1]
```diff
@@ -22,6 +22,9 @@
 #include "irods/irods_lexical_cast.hpp"
 #include "irods/irods_random.hpp"
 #include "irods/irods_at_scope_exit.hpp"
+#include "irods/voting.hpp"
+#include "irods/finalize_utilities.hpp"
+#include "irods/replica_proxy.hpp"
```
Comment (JustinKyleJames) [inline plugins/resources/src/compound.cpp:27]: These lines differ from 4-2-stable due to include directory changes.

File: plugins/resources/src/mockarchive.cpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #include "irods/irods_resource_redirect.hpp"
 #include "irods/irods_stacktrace.hpp"
 #include "irods/irods_server_properties.hpp"
+#include "irods/voting.hpp"
```
Comment (JustinKyleJames) [inline plugins/resources/src/mockarchive.cpp:17]: This line differs due to change in include directories.

**Comment by JustinKyleJames** [comment]
> @JustinKyleJames Please highlight what has changed between this PR and the cherry-pick.
> 
> Consider using the code review tab to select blocks of code that have changed.
I added in references that have changed.

**Review by korydraughn — COMMENTED** [review]
The changes to lib.py have been merged already.

Has this PR been rebased on top of main? If not, please rebase and consider whether the tests need to be run again.

**Comment by korydraughn** [comment]
I think the commit for 6690 just needs to be removed.

**Comment by JustinKyleJames** [comment]
> The changes to lib.py have been merged already.
> 
> Has this PR been rebased on top of main? If not, please rebase and consider whether the tests need to be run again.

When I opened this that had not yet been merged.  I went ahead and rebased on top of it.

A git diff shows the rebase is identical with the one before rebase except for the one line change that Marten made in 5b9cfb9f69e413512d7204421dc8cdb32ac5f1c1.

**Comment by korydraughn** [comment]
So it sounds like other than @MartinFlores751's db plugin commit, there's been no other changes made since you last tested this PR.

Is that correct?

**Comment by JustinKyleJames** [comment]
> So other than @MartinFlores751's db plugin commit, there's been no other changes made since you last tested this PR.
> 
> Is that correct?

Correct

**Review by korydraughn — APPROVED** [review]
Cool. Please add the pounds.

**Review by alanking — APPROVED** [review]

**Comment by JustinKyleJames** [comment]
> Cool. Please add the pounds.

Done

**Pull request merged**